### ISSUE TITLE
Long-running API call blocks others using same API key

### ISSUE THREAD

**Issue by qqmyers** [IQSS/dataverse#7061]
If I upload a large (e.g. >1GB) file and then call /api/admin/validateDataFileHashValue/<fileId> on it, which retrieves the file and recaculates its hash value, some other api calls including
/api/datasets/<id> and /api/dataverses/<id> are blocked until the hash recalculation finishes. (I don't think the file size is important beyond making the call last long enough to see the effect.)
However, viewing a dataset and adding new files in the web browser, or calling /api/admin/settings don't appear to be blocked. (The DVUploader is blocked as it calls the /api/datasets/<id> endpoint to find current files as well as retrieving presigned-upload-url(s) when in direct upload mode.)

FWIW: My testing is on a machine launched via ansible on EC2 with a manually configured AWS S3 store.

I don't yet understand what the bottleneck is, so any ideas would be welcome. The one commonality I can see so far is that api calls that require the apikey aren't working, whereas the /api/admin/settings call works due to allowing localhost calls. (Calling api/datasets/<id> on localhost:8080 with no apikey results in an immediate 401 response.)

**Comment by qqmyers** [comment]
After discussion in the DV Tech call (Thanks all!), and further testing based on the discussion, it appears that the call to authUser=userSvc.updateLastApiUseTime(authUser);  being done during the user lookup from an apiToken is what is making calls block. (At least removing it seems to resolve the issue in a branch where I'd changed the AuthetnticationServiceBean from @Singleton to @Stateless already, which didn't change things on it's own. Will try with an @Singleton bean tomorrow to confirm.) I also confirmed in tests that this is not related to S3 or hash validation - simply autheticating via api token and sleeping for 100 seconds blocks other calls and, conversely, removing the authentication step from hash validation, on S3 or files, is enough to make that non-blocking.

I'm not sure how to interpret the result yet except to assume that, without a separate transaction boundary, the transaction to update the lastApiUseTime is remaining open until the http call ends, which stops other calls (perhaps only for this user? all my testing has been two calls using the same API key - will try testing that as well.) If anyone has better insight into the cause or potential solutions, let me know.

**Comment by qqmyers** [comment]
Confirmed that this blocks only when using the same API Key - makes sense given that the db update is to the lastApiUseTime for that specific user. 

I have a simple fix - not sure if there are others that would be more performant: simply wrap the userSvc.updateLastApiUseTime(authUser) method with a RequiresNew transaction annotation. I've verified this works in one of my branches and will make a PR unless someone has another idea to try.
